client/src/hooks/useFetch.js

```diff
@@ -0,0 +1,42 @@
+import { useState, useEffect } from 'react';
+
+const useFetch = (url) => {
+  const [data, setData] = useState(null);
+  const [isPending, setIsPending] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const abortCont = new AbortController();
+
+    setTimeout(() => {
```

Comment:
Do we need timeout?
It makes sense if we are testing loading indicators. but should it be included in commits?